Backend/src/main/java/idealab/api/operations/PrintJobOperations.java

```diff
@@ -1,55 +1,67 @@
 package idealab.api.operations;
 
 import idealab.api.dto.GenericResponse;
+import idealab.api.dto.PrintJobDeleteRequest;
 import idealab.api.dto.PrintJobUpdateRequest;
+import idealab.api.model.Employee;
+import idealab.api.model.PrintStatus;
+import idealab.api.repositories.EmployeeRepo;
+import idealab.api.repositories.PrintStatusRepo;
 import org.springframework.stereotype.Component;
 
 @Component
 public class PrintJobOperations {
 
-//    EmployeeListRepo employeeListRepo;
-//    PrintStatusRepo printStatusRepo;
+    private EmployeeRepo employeeRepo;
+    private PrintStatusRepo printStatusRepo;
 
-    public GenericResponse updatePrintJob(PrintJobUpdateRequest dto)
+    public PrintJobOperations(EmployeeRepo employeeRepo, PrintStatusRepo printStatusRepo) {
+        this.employeeRepo = employeeRepo;
+        this.printStatusRepo = printStatusRepo;
+    }
+
+    public GenericResponse updatePrintJobStatus(Integer printId, PrintJobUpdateRequest dto)
     {
         GenericResponse response = new GenericResponse();
-
-        //Prep for when we get data models merged to master
-//        if(dto.isValidStatus())
-//        {
-//            //check if employee id is valid
-//            EmployeeList employeeList = employeeListRepo.getEmployeeListById(dto.getEmployeeId());
-//
-//            //check if print id is valid
-//            PrintStatus printStatus = printStatusRepo.getPrintStatusById(dto.getPrintStatusId());
-//
-//            if(employeeList != null && printStatus != null) {
-//                //Update print status
-//
-//
-//                //return success message
-//                response.setSuccess(true);
-//                response.setMessage("Print Job Updated");
-//            }
-//
-//        }
-//
-//        return response;
+        response.setSuccess(false);
+        response.setMessage("Print Job Status Update Failed");
 
         if(dto.isValidStatus())
         {
-            //check if employee id is valid
-            //check if print id is valid
-            //do any other logic to determine if the update is valid
-            response.setSuccess(true);
-            response.setMessage("Print Job Updated");
-        }
-        else
-        {
-            response.setSuccess(false);
-            response.setMessage("Invalid Status");
+            Employee employee = employeeRepo.getEmployeeById(dto.getEmployeeId());
+            PrintStatus printStatus = printStatusRepo.getPrintStatusById(printId);
+
+            if(employee != null && printStatus != null) {
+                printStatus = printStatusRepo.save(printStatus);
+                if(printStatus.getStatus().getName().equalsIgnoreCase(dto.getStatus())) {
+                    response.setSuccess(true);
```

Comment:
General note:  Need to discuss with front-end team how they want to update state.  If it is successful they can use what was sent to the backend, or we can send them the PrintModel with updated status as a data object that they can write back into state.  I am working on a PrintModel dto object that can be used to send back the data for all requests that we can use for all requests.